### ISSUE TITLE
CSS Selectors 4; :local-link

### ISSUE THREAD

**Issue by keithamus** [w3ctag/design-reviews#998]
こんにちは TAG-さん!

I'm requesting a TAG review of the `:local-link` pseudo class.

The :local-link pseudo-class allows authors to style [hyperlinks](https://drafts.csswg.org/selectors-4/#the-any-link-pseudo) based on the users current location within a site. It represents an element that is the source anchor of a hyperlink whose target’s absolute URL matches the element’s own document URL. If the hyperlink’s target includes a fragment URL, then the fragment URL of the current URL must also match; if it does not, then the fragment URL portion of the current URL is not taken into account in the comparison.

  - Explainer¹: https://drafts.csswg.org/selectors-4/#the-local-link-pseudo
  - Specification: https://drafts.csswg.org/selectors-4/#the-local-link-pseudo
  - WPT Tests: (Not yet, will live in https://github.com/web-platform-tests/wpt/tree/master/css/selectors)
  - User research:
  - Security and Privacy self-review²:
  - GitHub repo:
  - Primary contacts:
      - Keith Cirkel (@keithamus), Microsoft/GitHub, Implementer
      - Lea Verou (@LeaVerou), W3C Invited Expert, CSSWG Invited Expert
      - (@fantasai), Apple, CSSWG Spec writer
  - Organization/project driving the specification: Already specified
  - Multi-stakeholder support³:
    - Chromium comments:
    - Mozilla comments: https://github.com/mozilla/standards-positions/issues/862
    - WebKit comments: https://github.com/WebKit/standards-positions/issues/405
    - {{...include feedback/review from developers, implementers, civil society, and others}}
  - Status/issue trackers for implementations⁴:

Further details:

  - [x] I have reviewed the TAG's [Web Platform Design Principles](https://www.w3.org/TR/design-principles/)
  - Relevant time constraints or deadlines: I'll be working on prototypes for browsers this week.
  - The group where the work on this specification is currently being done: CSSWG
  - The group where standardization of this work is intended to be done (if different from the current group): CSSWG
  - Major unresolved issues with or opposition to this specification: None
  - This work is being funded by: Me, or I guess GitHub, but they allow me to contribute as an independant.

You should also know that...

I'll be implementing a prototype this week in Chrome/Firefox/WebKit.

**Comment by jyasskin** [comment]
Thanks! My personal review is in https://github.com/w3c/csswg-drafts/issues/10975, but that's not the full TAG's opinion. The use case, at least, seems great.

**Comment by LeaVerou** [comment]
Agree with @jyasskin about the name, but the use case is a **huge** pain point. It is *very* nontrivial to implement with JS, and comes up in literally every navigation menu. The lack of this made teaching web technologies significantly harder.

**Comment by jyasskin** [comment]
The TAG discussed this as a whole and agreed that we're enthusiastic about the use case being solved, and want it to be solved [with more precise names](https://github.com/w3c/csswg-drafts/issues/10975). Thank you!
